File: 1470-shuffle-the-array/1470-shuffle-the-array.c
int* shuffle(int* nums, int numsSize, int n, int* returnSize){
     *returnSize = numsSize;
    int* ans = (int*) malloc(numsSize * sizeof(int));
    int k=0;
    for(int i=0;i<numsSize;i++){
        if(i%2==0){
            ans[i]=nums[k++];
        }
        else{
            ans[i]=nums[n++];
        }
    }
    return ans;
}

File: 1470-shuffle-the-array/1470-shuffle-the-array.java
class Solution {
    public int[] shuffle(int[] nums, int n) {
        int[] arr = new int[2*n];
        int i = 0, j = n, k = 0;
        
        while(k < 2*n)
        {
            if(k % 2 == 0)
                arr[k] = nums[i++];
            else
                arr[k] = nums[j++];
            k++;
        }
        return arr;
    }
}